### PR TITLE
Fix # 273

### PR DESCRIPTION
**Pull request by noumanahmads**
- squash multiple contract calls
- created a new function in` fixed-weighted-pool `

### CODE REVIEW

**Comment by fiftyeightandeight** [comment]
Please update your branch to the latest `dev` and go through all pool contracts to apply the same squash to where applicable. For example:

https://github.com/alexgo-io/alex-v1/blob/aa4b57cac398ac558f24a4d8c210c9aed948cd73/clarity/contracts/pool/collateral-rebalancing-pool.clar#L443-L445

https://github.com/alexgo-io/alex-v1/blob/aa4b57cac398ac558f24a4d8c210c9aed948cd73/clarity/contracts/pool/collateral-rebalancing-pool.clar#L483-L485

https://github.com/alexgo-io/alex-v1/blob/aa4b57cac398ac558f24a4d8c210c9aed948cd73/clarity/contracts/pool/collateral-rebalancing-pool.clar#L494-L497

https://github.com/alexgo-io/alex-v1/blob/aa4b57cac398ac558f24a4d8c210c9aed948cd73/clarity/contracts/pool/collateral-rebalancing-pool.clar#L747-L750

File: clarity/contracts/pool/fixed-weight-pool.clar [hunk 1]
```diff
@@ -1109,3 +1111,19 @@
    )
  )
 )
+
+(define-public (get-weighted-dy (token <ft-trait>) (collateral <ft-trait>) (weight-x uint) (weight-y uint) (dx-to-dy uint) (min-dx (optional uint)) (min-dy (optional uint)))
```
Comment (fiftyeightandeight) [inline clarity/contracts/pool/fixed-weight-pool.clar:1115]: `get-weighted-dy` and `get-balance-x-to-y` are almost identical. Can we please re-write them as

`(define-public (swap (token-x-trait <ft-trait>) (token-y-trait <ft-trait>) (weight-x uint) (weight-y uint) (dx uint) (min-dy (optional uint))`

**Review by fiftyeightandeight — CHANGES_REQUESTED** [review]

File: clarity/contracts/pool/fixed-weight-pool.clar (hunk 1 above)
Comment (noumanahmads) [reply clarity/contracts/pool/fixed-weight-pool.clar]: done

**Comment by fiftyeightandeight** [comment]
Hi @noumanahmads , I resolved the conflicts, but would like to wait for https://github.com/alexgo-io/alex-v1/pull/319 to merge before merging this PR.

**Pull request merged**